Sergio Arrocha y Osmar Del Rosario/otrolabconsergio.c

```diff
@@ -0,0 +1,33 @@
+#include<stdio.h>
+#define MAT 3
+void main(){
+int f, c, matriz[MAT][MAT];
+for (f=0;f<MAT;f++){
+for (c=0;c<MAT;c++){
+	printf("Introduzca el valor en la posicion ");
+	printf("[%d][%d]:", f,c);
+	scanf("%d",& matriz[f][c]);
+	printf("\n");
+}
+}	
+for (f=0;f<MAT;f++){
+for (c=0;c<MAT;c++){
+	printf("%d \t", matriz[f][c]);	
+}
+printf("\n");
+}
+printf("\n");
+
+while(matriz[f][c]==1){
+
+
+if((f==c)!=1){

```

Comment:
La forma como calculas el valor de la matriz identidad esta equivocado
Una forma sencilla es con dos ciclos for, verificando que solo la diagonal tenga valor de 1 y el resto de los valores sean 0